### PR TITLE
treewide: apply deadnix and statix

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [ ] Tested the changes in your own NixOS Configuration
- [ ] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

**Comment by Mic92** [comment]
bors merge

File: dell/latitude/3340/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, pkgs, ... }:
+{ ... }:
```
Comment (SuperSandro2000) [inline dell/latitude/3340/default.nix:1]: like here

File: toshiba/swanky/default.nix [hunk 2]
```diff
@@ -20,7 +20,7 @@ in
   system.replaceRuntimeDependencies = [{
     original = pkgs.alsa-lib;
 
-    replacement = pkgs.alsa-lib.overrideAttrs (super: {
+    replacement = pkgs.alsa-lib.overrideAttrs (_super: {
```
Comment (SuperSandro2000) [inline toshiba/swanky/default.nix:23]: This also looks like a mistake

File: raspberry-pi/4/pkgs-overlays.nix [hunk 3]
```diff
@@ -1,9 +1,9 @@
-{ config, lib, pkgs, modulesPath, ... }:
+{ config, lib, ... }:
 let
   cfg = config.hardware.raspberry-pi."4".apply-overlays-dtmerge;
-  dt_ao_overlay = (final: prev: {
-    deviceTree.applyOverlays = (prev.callPackage ./apply-overlays-dtmerge.nix { });
-  });
+  dt_ao_overlay = _final: prev: {
```
Comment (SuperSandro2000) [inline raspberry-pi/4/pkgs-overlays.nix:4]: same here

**Comment by SuperSandro2000** [comment]
> bors merge

I don't think we need bors. This repo has such a low volume of PRs and most don't touch code that is related to each other.

File: toshiba/swanky/default.nix (hunk 2 above)
Comment (Mic92) [reply toshiba/swanky/default.nix:23]: Not a mistake. deadnix renames variables if they are unused.

**Comment by Mic92** [comment]
> > bors merge
> 
> I don't think we need bors. This repo has such a low volume of PRs and most don't touch code that is related to each other.

The test suite takes very long as it has to evaluate many nixos configuration. So we need bors. It enables reviewers to review pull requests much faster since I don't have to wait for CI to finish.

File: apple/macbook-pro/10-1/default.nix [hunk 4]
```diff
@@ -1,5 +1,3 @@
-{ lib, pkgs, ... }:
```
[the remark this answers is not in the document]
Comment (Mic92) [reply apple/macbook-pro/10-1/default.nix:1]: No. `{}` is a valid nixos module.

File: toshiba/swanky/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply toshiba/swanky/default.nix:23]: Thats new? I thought it just replaced them with _

**Comment by SuperSandro2000** [comment]
If bors just waits until CI is done and then merges it that is totally fine and a really nice addition IMO but the merge train part of it that rebases PRs always on the latest commit is really wasteful for most PRs especially if they touch parts that are so far apart that they possible can't influence each other.

**Comment by Mic92** [comment]
Mhm. So far we seem to not hit any scalability issues - our open PR count is quite low. Note that bors-ng is also designed to batch PRs once there are enough in a queue. This works quite well on big projects with slow CIs such as rust.

**Comment by SuperSandro2000** [comment]
Here that is not a problem because everything runs on actions but in nixpkgs ofborg could really struggle with this.

**Comment by Mic92** [comment]
I would really love to have a `Merge when CI is green` button in nixpkgs...

**Comment by sambacha** [comment]
There is a GitHub Merge Queue that can potentially replace the usage of Bors btw now https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue

**Comment by Mic92** [comment]
Yeah, unfortunately you need to pay for github for this feature, which is not feasible for a big project like nixpkgs.

**Comment by Mic92** [comment]
Maybe they will make available later for opensource projects?

> Note: The pull request merge queue feature is currently in limited public beta and subject to change.

**Pull request merged**